drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,735 @@
+    if (dev->settings.lora.freq_hop_on) {
+        tmp |= SX127X_RF_LORA_PLLHOP_FASTHOP_ON;
+        sx127x_reg_write(dev, SX127X_REG_LR_PLLHOP, tmp);
+        sx127x_reg_write(dev, SX127X_REG_LR_HOPPERIOD, hop_period);
+    }
+}
+
+bool  sx127x_get_implicit_header_mode(sx127x_t *dev)
+{
+    return dev->settings.lora.implicit_header_mode;
+}
+
+void sx127x_set_implicit_header_mode(sx127x_t *dev, bool implicit)
+{
+    dev->settings.lora.implicit_header_mode = implicit;
+
+    uint8_t config1_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1);
+    switch (dev->driver_type) {
+    case SX1276_TYPE:
+        config1_reg &= SX1276_RF_LORA_MODEMCONFIG1_IMPLICITHEADER_MASK;
+        config1_reg |= implicit;
+        break;
+    case SX1272_TYPE:
+        config1_reg &= SX1272_RF_LORA_MODEMCONFIG1_IMPLICITHEADER_MASK;
+        config1_reg |= implicit << 2;
+        break;
+    default:
+        DEBUG("Unknown device, %d", dev->driver_type);
+        break;
+    }
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, config1_reg);
+}
+
+uint8_t sx127x_get_payload_length(sx127x_t *dev)
+{
+    return dev->settings.lora.payload_len;
+}
+
+void sx127x_set_payload_length(sx127x_t *dev, uint8_t len)
+{
+    if (dev->settings.lora.implicit_header_mode) {
+        dev->settings.lora.payload_len = len;
+        sx127x_reg_write(dev, SX127X_REG_LR_PAYLOADLENGTH, len);
+    }
+}
+
+static inline uint8_t sx127x_get_pa_select(uint32_t channel)
+{
+    if (channel < LORA_RF_MID_BAND_THRESH) {
+        return SX127X_RF_PACONFIG_PASELECT_PABOOST;
+    }
+    else {
+        return SX127X_RF_PACONFIG_PASELECT_RFO;
+    }
+}
+
+uint8_t sx127x_get_power(sx127x_t *dev)
+{
+    return dev->settings.lora.power;
+}
+
+void sx127x_set_power(sx127x_t *dev, uint8_t power)
+{
+    dev->settings.lora.power = power;
+
+    uint8_t pa_config = sx127x_reg_read(dev, SX127X_REG_PACONFIG);
+    uint8_t pa_dac = sx127x_reg_read(dev, SX127X_REG_PADAC);
+
+    pa_config = ((pa_config & SX127X_RF_PACONFIG_PASELECT_MASK) |
+                 sx127x_get_pa_select(dev->settings.channel) << 7);
+    /* max power is 14dBm */
+    pa_config = (pa_config & SX127X_RF_PACONFIG_MAX_POWER_MASK) | (0x05 << 4);
+
+    sx127x_reg_write(dev, SX127X_REG_PARAMP, SX127X_RF_PARAMP_0050_US);
+
+    if ((pa_config & SX127X_RF_PACONFIG_PASELECT_PABOOST)
+        == SX127X_RF_PACONFIG_PASELECT_PABOOST) {
+        if (power > 17) {
+            pa_dac = ((pa_dac & SX127X_RF_PADAC_20DBM_MASK) |
+                      SX127X_RF_PADAC_20DBM_ON);
+        } else {
+            pa_dac = ((pa_dac & SX127X_RF_PADAC_20DBM_MASK) |
+                      SX127X_RF_PADAC_20DBM_OFF);
+        }
+        if ((pa_dac & SX127X_RF_PADAC_20DBM_ON) == SX127X_RF_PADAC_20DBM_ON) {
+            if (power < 5) {
+                power = 5;
+            }
+            if (power > 20) {
+                power = 20;
+            }
+
+            pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
```

Comment:
Need to rewrite this (and others `pa_config` within this function) to match SX1272 register which is different from SX1276